server/server.go

```diff
@@ -1062,6 +1062,11 @@ func addFlushErrDetailsOrReturn(s *status.Status, d *spb.FlushResponseError) err
 // validating that the election parameters are consistent, and correct, and that the Flush should be
 // completed.
 func (s *Server) checkFlushRequest(req *spb.FlushRequest) error {
+	if req.GetNetworkInstance() == nil {
```

Comment:
I'd suggest doing this in two checks:

```golang
switch {
  case req == nil:
     // return error about a nil message
  case req.GetNetworkInstance() == nil:
     // return error about a nil network-instance
}
```

I think that the two cases (`nil` in the input vs. `nil` in the network-instance) probably have different provenance in terms of root cause, such that it's likely worth differentiating them.

Reply:
(Especially since the error codes here should probably be different.)

Reply:
Despite `req==nil` requires another error code to be defined (not a big deal), my thought is that `req` can not be `nil` is **only** because `req.NetworkInstance` can not be `nil`. Hence, I only need to check `req.GetNetworkInstance() == nil`, right?

Reply:
Right, I don't think the gRPC transport layer would ever call the server stub with a nil request proto; req could be an empty proto but not nil.

Reply:
IMHO, if it is possible to be called this way at all, then it's worth handling and not having a NPE. w.r.t whether we need a new error code, you can just return a generic `codes.Internal` for these kind of "there is something seriously wrong" issue.

Reply:
Done.

Reply:
@robshakir